src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,98 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static RegistrationService service;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        service = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setLogin("Jack");
+        user.setPassword("qwertyu");
+        user.setAge(20);
+    }
+
+    @Test
+    void addNull_NotOk() {
+        user = null;
+        assertThrows(RuntimeException.class, () ->
+                service.register(user), "User can't be null");
+    }
+
+    @Test
+    void addNegativeAge_NotOk() {
+        user.setAge(-1);
+        assertThrows(RuntimeException.class, () ->
+                 service.register(user), "Age can't be negative");
+    }
+
+    @Test
+    void addNullLogin_NotOk() {
+        user.setLogin(null);
+        assertThrows(RuntimeException.class, () ->
+                service.register(user), "Login can't be null");
+    }
+
+    @Test
+    void addNullPassword_NotOk() {
+        user.setPassword(null);
+        assertThrows(RuntimeException.class, () ->
+                service.register(user), "Password can't be null");
+    }
+
+    @Test
+    void userIsTooYoung_NotOk() {
+        user.setAge(MIN_AGE - 1);
+        assertThrows(RuntimeException.class, () ->
+                service.register(user), "User is too young: age" + user.getAge());
+    }
+
+    @Test
+    void registeredSameUser_notOk() {
+        service.register(user);
```

Comment:
You can’t use method register() here because you testing this method. You need to add a user to the base like this: Storage.people.add().
And then register the user with the same email using register() method expecting an exception. Check it everywhere